### PR TITLE
feat(java): allow dynamic java install

### LINKED ISSUE #22: Use adopt openjdk
> We should consider using [AdoptOpenJDK](adoptopenjdk.net) for building all java version
> 
> api: https://api.adoptopenjdk.net/swagger-ui/
> 
> https://api.adoptopenjdk.net/v3/info/available_releases
> 
> ```json
> {
>   "available_lts_releases": [
>     8,
>     11
>   ],
>   "available_releases": [
>     8,
>     9,
>     10,
>     11,
>     12,
>     13,
>     14
>   ],
>   "most_recent_feature_release": 14,
>   "most_recent_lts": 11
> }
> ```
> 
> https://api.adoptopenjdk.net/v3/info/release_versions?page=0&page_size=10&release_type=ga&sort_order=DESC&vendor=adoptopenjdk
> 
> ```json
>     {
>       "adopt_build_number": 1,
>       "build": 9,
>       "major": 13,
>       "minor": 0,
>       "openjdk_version": "13.0.1+9",
>       "security": 1,
>       "semver": "13.0.1+9.1"
>     },
>     {
>       "adopt_build_number": 1,
>       "build": 33,
>       "major": 13,
>       "minor": 0,
>       "openjdk_version": "13+33",
>       "security": 0,
>       "semver": "13.0.0+33.1"
>     },
> ```

---
https://github.com/renovatebot/docker-buildpack/issues/33#issue-620688908

### PR DESCRIPTION
**Pull request by viceice**
ref #16 

closes #22

### CODE REVIEW

**Comment by viceice** [comment]
Curretnly only one version of v8, v11 and v16 is available. So i think we need to add the adopt api to renovate too.

**Comment by rarkins** [comment]
I don't see any `USER X` changes here. Is it now installable as user instead of root?

**Comment by viceice** [comment]
> I don't see any `USER X` changes here. Is it now installable as user instead of root?

Yes, need to update tests. 🤗

File: test/java/Dockerfile [hunk 1]
```diff
@@ -116,6 +133,7 @@ RUN set -ex; \
 
 RUN ls -la ~
 
+
```
Comment (viceice) [inline test/java/Dockerfile:136]:
```suggestion
```

**Pull request merged**